yargs.js

```diff
@@ -1010,7 +1018,14 @@ function Yargs (processArgs, cwd, parentRequire) {
     args = args || processArgs
 
     options.__ = y18n.__
-    options.configuration = pkgUp()['yargs'] || {}
+    options.configuration = self.getParserConfiguration()
+
+    // Deprecated
+    let pkgConfig = pkgUp()['yargs']
```

Comment:
I like @bcoe's suggestion to use [`util.deprecate()`](https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code) here. It'll require a minor extraction of the logic, but that's arguably a good thing anyway.

```js
const util = require('util');

const getPkgConfig = util.deprecate(() => pkgUp()['yargs'], 'Configuring yargs through package.json is deprecated...');

// etc etc
```

We can test it by listening to the [warning](https://nodejs.org/api/process.html#process_event_warning) event on `process`:
```js
let called = false
process.once('warning', () => {
  called = true
})
// do the thing
// assert called === true ?
```

Reply:
Wouldn't this emit the warning whether package.json contains yargs configuration tokens or not ? This deprecates the function that checks for package.json configuration, I think it makes more sense to only warn when the deprecated feature is actually used.

Reply:
Oh, yeah, you're right, nevermind `util.deprecate`